### PR TITLE
Change kernel name toolbar indicator to button that switches kernel

### PR DESCRIPTION
**Pull request by markroth8**
Fixes second part of #919.

Changes the Kernel name in the toolbar to a button that, when clicked, displays the "Switch Kernel" dialog.

Needed to add a spacer because the kernel name label presently takes up the remaining whitespace on the toolbar which would make the button look huge.

### CODE REVIEW

**Comment by blink1073** [comment]
Nice!  @eskirk, I prefer this to the context menu approach because it is more discoverable: see hover screenshot:

<image src="https://cloud.githubusercontent.com/assets/2096628/25965236/fefe9db2-364c-11e7-9bf7-a6e2e5d711f0.png" width=200>

**Comment by blink1073** [comment]
Ah, we have a test failure: https://travis-ci.org/jupyterlab/jupyterlab/jobs/231263945#L4138
This is the test: https://github.com/jupyterlab/jupyterlab/blob/f13cad7b57bb6ba1be4784a8010640c6276d905a/test/src/notebook/default-toolbar.spec.ts#L264.

**Comment by ellisonbg** [comment]
Sitting here working with @markroth8 - I too think I prefer the button to the dropdown...

File: packages/apputils/src/toolbar.ts [hunk 1]
```diff
@@ -228,6 +233,19 @@ namespace Toolbar {
...
 
   /**
+   * Create a toolbar spacer item.
+   *
+   * #### Notes
+   * It is a flex spacer that separates the left toolbar items
+   * from the right toolbar items.
+   */
+  export
+  function createSpacerItem(session: IClientSession): Widget {
```
Comment (ellisonbg) [inline packages/apputils/src/toolbar.ts]: I don't think the spacer needs the session passed in.

**Comment by markroth8** [comment]
That teaches me to assume such a change can't possibly break tests!

File: packages/apputils/src/toolbar.ts (hunk 1 above)
Comment (markroth8) [inline packages/apputils/src/toolbar.ts]: True.

**Comment by blink1073** [comment]
Thanks again!

**Pull request merged**